lib/Models/Workbench.ts

```diff
@@ -167,14 +166,15 @@ export default class Workbench {
    *
    * @param item The item to add to or remove from the workbench.
    */
-  public async add(item: BaseModel | BaseModel[]): Promise<void> {
+  public async add(
+    item: BaseModel | BaseModel[],
+    index: number = 0
+  ): Promise<void> {
     if (Array.isArray(item)) {
-      await Promise.all(item.reverse().map(i => this.add(i)));
+      await Promise.all(item.reverse().map(i => this.add(i, index)));
       return;
     }
 
-    this.insertItem(item);
-
```

Comment:
Moved this call to later, because if the item is a reference, we need to load it first.